pkg/action/upgrade.go

```diff
@@ -427,5 +427,5 @@ func recreate(cfg *Configuration, resources kube.ResourceList) error {
 
 func objectKey(r *resource.Info) string {
 	gvk := r.Object.GetObjectKind().GroupVersionKind()
-	return fmt.Sprintf("%s/%s/%s/%s", gvk.GroupVersion().String(), gvk.Kind, r.Namespace, r.Name)
+	return fmt.Sprintf("%s/%s/%s", gvk.Kind, r.Namespace, r.Name)
```

Comment:
I considered this as well. Unfortunately ignoring group and version would allow collisions with custom resources from different API groups. Perhaps this is something we might be able to tolerate and call out in docs, but I'm not sure.

@bacongobbler @hickeyma any thoughts?

Reply:
If we could exclude `gvk.GroupVersion().String()` for native k8s resource types but _not_ for custom resources, then I think this might be fine.

Reply:
This would be a change in Helm's existing behaviour, right? I don't think we can remove this behaviour if we made the assumption that we would not allow collisions in previous versions of Helm 3.

Reply:
We also do not have any assurance from Upstream Kubernetes that they won't at some point decide to have multiple kinds with the same names that have different group/version.